src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -41,7 +43,12 @@ public FindCommand parse(String args) throws ParseException {
                 ? argMultimap.getValue(PREFIX_PHONE).get().trim().split("\\s+")
                 : null;
         String[] addressKeywords = argMultimap.isPresent(PREFIX_ADDRESS)
-                ? argMultimap.getValue(PREFIX_ADDRESS).get().trim().split("_")
+                ? Arrays.stream(argMultimap.getValue(PREFIX_ADDRESS).get().trim().split("_"))
+                    .map(String::trim).toArray(String[]::new)
+                : null;
+        String[] tagKeywords = argMultimap.isPresent(PREFIX_TAG)
+                ? Arrays.stream(argMultimap.getValue(PREFIX_TAG).get().trim().split("_"))
+                    .map(String::trim).toArray(String[]::new)
```

Comment:
nice work on this portion of the code, to split the input based on "_" for multiple addresses and tags